src/jit/morph.cpp

```diff
@@ -4395,24 +4429,26 @@ void Compiler::fgMorphSystemVStructArgs(GenTreeCall* call, bool hasStructArgumen
                         // Create LCL_FLD for each eightbyte.
                         argListCreated = true;
 
+                        // First field
+                        arg->AsLclFld()->gtFieldSeq = FieldSeqStore::NotAField();
+                        arg->gtType =
+                            GetTypeFromClassificationAndSizes(fgEntryPtr->structDesc.eightByteClassifications[0],
+                                                              fgEntryPtr->structDesc.eightByteSizes[0]);
+                        GenTreeFieldList* fieldList =
+                            new (this, GT_FIELD_LIST) GenTreeFieldList(arg, 0, originalType, nullptr);
+                        fieldList->gtType          = originalType; // Preserve the type. It is a special case.
+                        arg = fieldList;
```

Comment:
That might be possible, though it might add complexity because currently the items in the `GT_FIELD_LIST` have the same size & type as the field.

---
In reply to: [79517793](https://github.com/dotnet/coreclr/pull/7252#discussion_r79517793) [](ancestors = 79517793)